spec/models/qc_reception_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe QcReception do
+  describe '#create' do
+    before do
+      create(:qc_assay_type, key: 'sheared_femto_fragment_size', label: 'Sheared Femto Fragment Size (bp)', used_by: 2, units: 'bp')
+    end
+
+    let(:qc_reception) { build(:qc_reception) }
+
+    it 'is possible to create a new record' do
+      expect do
+        described_class.create!(
+          source: qc_reception.source,
+          qc_results_list: qc_reception.qc_results_list
+        )
+      end.to change(described_class, :count).by(1)
+    end
+
+    describe '#validates_nested' do
+      it 'errors if missing required source field' do
+        expect do
+          described_class.create!(qc_results_list: qc_reception.qc_results_list)
+        end.to raise_error(ActiveRecord::RecordInvalid)
+      end
+
+      it 'errors if missing required qc_results_list' do
+        expect do
+          described_class.create!(source: qc_reception.source)
+        end.to raise_error(ActiveRecord::RecordInvalid)
+      end
+    end
+
+    context 'when the qc_results_list is an empty object' do
+      let(:qc_reception) { build(:qc_reception, qc_results_list: [{}]) }
+
+      it 'raises validation error' do
+        expect do
+          described_class.create!(
+            source: qc_reception.source,
+            qc_results_list: qc_reception.qc_results_list
+          )
+        end.to raise_error(ActiveRecord::RecordInvalid)
+      end
+    end
+  end
+
+  describe '#check delegation' do
```

Comment:
I don't think these tests are necessary as they are testing implementation rather than behaviour. You would soon know if the method was delegated as you would get a NoMethodError

Reply:
Removed them